connect_ext_ppr/constants.py

```diff
@@ -535,3 +535,38 @@
 """
 SUMMARY_TEMPLATE = """Summary:
 {summary}"""
+
+
+CONFIGURATION_SCHEMA_TEMPLATE = """
```

Comment:
Why do you need a template in this case? I think it is already possible to have dynamic fields as keys

```
def configuration_schema(product_id):
  return {
     product_id: {<schema>},
     "required": [product_id]
  }

```

Reply:
Because it will split schema into 2 parts:
**Outer**
```
def configuration_schema(product_id):
  return {
    "type": "object",
    "properties": {
        "hierarchical_files_data": {
            "type": "object",
            "properties": {
                product_id: **inner_schema,
                "required": [product_id],
            },
            "required": ["{product_id}"],
        },
    },
    "required": ["hierarchical_files_data"],
}
```
**Inner**
```
inner_schema = {
    "type": "object",
    "properties": {
        "product_level": {
            "type": "object",
            "properties": {
                "ResourceCategories": {
                    "type": "object",
                    "anyOf": [
                        {"required": ["Name_EN"]},
                        {"required": ["Name_en"]}
                    ]
                }
            },
            "required": ["ResourceCategories"]
        }
    },
    "required": ["product_level"]
}
```

My idea was to keep schema as a constant with other constants. If we want to use approach with the function, we may put all the schema inside the function, so it all will be stored in one place.
I think that this template is the most visual. Also we may store it as a dict and convert to string and back when it is required.

Reply:
For it is a little bit complex... Inside the library it is always a dict